### PR TITLE
[CELEBORN-546][FLINK] Use autoIncrement partitionId replace encode(mapId, attemptId) for generating partitionId

### PR DESCRIPTION
**Pull request by RexXiong**
…pId, attemptId) for generating partitionId

### What changes were proposed in this pull request?
Use autoIncrement partitionId replace encode(mapId, attemptId) for generating partitionId

### Why are the changes needed?
encode(mapId, attemptId) will cause attemptId is limit to 255

### Does this PR introduce _any_ user-facing change?
yes

### How was this patch tested?
UT & TPCDS

### CODE REVIEW

File: common/src/main/java/org/apache/celeborn/common/protocol/PartitionLocation.java [hunk 1]
```diff
@@ -286,13 +285,9 @@ public String toString() {
       peerAddr = peer.hostAndPorts();
     }
     return "PartitionLocation["
-        + "\n  id(rawId-attemptId)-epoch:"
+        + "\n  id-epoch:"
```
Comment (zhongqiangczq) [inline common/src/main/java/org/apache/celeborn/common/protocol/PartitionLocation.java:288]: if we want to look at shufflewring filename of one map task, it's not directly to build the relation between filename (  id + "-" + epoch + "-" + mode.mode) and the mapid

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala [hunk 2]
```diff
@@ -957,22 +956,13 @@ class PushDataHandler extends BaseMessageHandler with Logging {
       callback: RpcResponseCallback,
       wrappedCallback: RpcResponseCallback): Boolean = {
```
Comment (waitinfuture) [inline worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala:957]: wrappedCallback is not used. BTW, ```locationIsNull```, ```handleRpcRequestCore``` are only for MapPartition, I think better to refactor in the future.

File: client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleMaster.java [hunk 3]
```diff
@@ -238,18 +240,19 @@ public void close() throws Exception {
     ThreadUtils.shutdownExecutors(10, executor);
   }
 
-  private ShuffleTask encodeExternalShuffleTask(
+  private ShuffleResourceDescriptor genShuffleResourceDescriptor(
```
[the remark this answers is not in the document]
Comment (RexXiong) [reply client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleMaster.java:239]: No, same like attemptId, so i remove mapattemptId->attemptIdIndex map as same

File: worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala (hunk 2 above)
Comment (RexXiong) [reply worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala:957]: > wrappedCallback is not used. BTW, `locationIsNull`, `handleRpcRequestCore` are only for MapPartition, I think better to refactor in the future.

Agree.

File: client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleDescriptor.java [hunk 4]
```diff
@@ -58,6 +58,10 @@ public JobID getJobId() {
     return jobId;
   }
 
+  public String getShuffleId() {
```
Comment (waitinfuture) [inline client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleDescriptor.java:61]: This method is not used

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

File: common/src/main/java/org/apache/celeborn/common/protocol/PartitionLocation.java (hunk 1 above)
Comment (RexXiong) [reply common/src/main/java/org/apache/celeborn/common/protocol/PartitionLocation.java:288]: It's partial right, as the `<int> mapId `also comes from `LifecycleManager`. Only `LifecycleManager` has the knowledge which partition belongs to the actual resultPartition.

**Pull request merged**